util/src/main/java/com/redhat/lightblue/util/metrics/DropwizardRequestMetrics.java

```diff
@@ -0,0 +1,135 @@
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.redhat.lightblue.util.metrics;
+
+import com.codahale.metrics.Counter;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.Timer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.UndeclaredThrowableException;
+import java.util.Objects;
+
+import static com.codahale.metrics.MetricRegistry.name;
+
+public class DropwizardRequestMetrics implements RequestMetrics {
+    
+    private final Logger LOGGER = LoggerFactory.getLogger(DropwizardRequestMetrics.class);
+
+    private static final String API = "api";
+
+    private final MetricRegistry metricsRegistry;
+
+    public DropwizardRequestMetrics(MetricRegistry metricRegistry) {
+        metricsRegistry = metricRegistry;
+    }
+
+    /**
+     * Create exception namespace for metric reporting based on exception name
+     * 
+     */
+    private static String errorNamespace(String metricNamespace, Throwable exception) {
+        Class<? extends Throwable> actualExceptionClass = unravelReflectionExceptions(exception);
+        return name(metricNamespace, "requests", "exception", actualExceptionClass.getSimpleName());
+    }
+
+    /**
+     * Get to the cause we actually care about in case the bubbled up exception is a
+     * higher level framework exception that encapsulates the stuff we really care
+     * about.
+     * 
+     */
+    private static Class<? extends Throwable> unravelReflectionExceptions(Throwable e) {
+        while (e.getCause() != null
+                && (e instanceof UndeclaredThrowableException || e instanceof InvocationTargetException)) {
+            e = e.getCause();
+        }
+        return e.getClass();
+    }
+
+    @Override
+    public Context startEntityRequest(String operation, String entity, String version) {
+        return new DropwizardContext(name(API, operation, entity, version));
+    }
+
+    @Override
+    public Context startStreamingEntityRequest(String operation, String entity, String version) {
+        return new DropwizardContext(name(API, "stream", operation, entity, version));
+    }
+    
+    @Override
+    public Context startLockRequest(String lockOperation, String domain) {
+        return new DropwizardContext(name(API, "lock", domain, lockOperation));
+    }
+
+    @Override
+    public Context startHealthRequest() {
+        return new DropwizardContext(name(API, "checkHealth"));
+    }
+
+    @Override
+    public Context startDiagnosticsRequest() {
+        return new DropwizardContext(name(API, "checkDiagnostics"));
+    }
+
```

Comment:
They were initially included for consistency's sake, and even though I am not exactly sure how these will be used yet, I think we should leave them in there.   I can't think of any uses for the diagnostics one at the moment, however perhaps the health one can serve as a baseline /control for other monitored find requests, as it does a find on an entity with minimal data and few clients.  I guess if we find they're not useful at all we can just remove these later.